test/server.go

```diff
@@ -15,17 +15,21 @@ import (
 
 var mu sync.Mutex
 
-// CoreDNSServer returns a CoreDNS test server. It just takes a normal Corefile as input.
-func CoreDNSServer(corefile string) (*caddy.Instance, error) {
+// CoreDNSServerWithLog returns a CoreDNS test server. It just takes a normal Corefile as input.
+func CoreDNSServerWithLog(corefile string) (*caddy.Instance, error) {
 	mu.Lock()
 	defer mu.Unlock()
 	caddy.Quiet = true
 	dnsserver.Quiet = true
-	log.SetOutput(ioutil.Discard)
-
 	return caddy.Start(NewInput(corefile))
 }
 
+// CoreDNSServer returns a CoreDNS test server. It just takes a normal Corefile as input and discards logging.
+func CoreDNSServer(corefile string) (*caddy.Instance, error) {
+	log.SetOutput(ioutil.Discard)
```

Comment:
Just remove this from the original function (or does that make the tests to spammy?) and only leave `CoreDNSServer`

Reply:
Done.